### PR TITLE
Add parWrapChecks to HealthReporter, refactor results reduction to a single method

### LINKED ISSUE #73: HealthReporter should provide a way of run-ing checks in parallel using cats.parallel machinery
Useful for cases where reporter health checks are completely independent and one does not need to create a wrapper health check using parallel combinators.

### PR DESCRIPTION
**Pull request by ragb**
fixes #73

### CODE REVIEW

File: modules/core/src/main/scala/sup/data/HealthReporter.scala [hunk 1]
```diff
@@ -24,10 +24,25 @@ object HealthReporter {
...
+  }
+
+    /**
+    * Constructs a healthcheck from a non-empty structure G of healthchecks that are run in parallel.
+    * The status of the whole check is determined using the results of given checks and the semigroup of Health.
+    *
+    * e.g. if all checks need to return Healthy for the whole thing to be healthy, use [[Health.allHealthyCommutativeMonoid]].
+    */
+  def parWrapChecks[F[_]: Apply, G[_]: NonEmptyTraverse, H[_]: Reducible, M[_]](checks: G[HealthCheck[F, H]])(
+    implicit M: Semigroup[Health], P: Parallel[F, M]): HealthReporter[F, G, H] = HealthCheck.liftF {
```
Comment (kubukoz) [inline modules/core/src/main/scala/sup/data/HealthReporter.scala]: NonEmptyParallel should be enough. Additionally, if sup already has cats-par in dependencies (I think it does), use NonEmptyPar.

File: modules/core/src/main/scala/sup/data/HealthReporter.scala [hunk 2]
```diff
@@ -24,10 +24,25 @@ object HealthReporter {
...
+
+    Parallel.parNonEmptyTraverse(checks)(_.check).map(reduceResults[G, H] _)
```
Comment (kubukoz) [inline modules/core/src/main/scala/sup/data/HealthReporter.scala]: We could use syntax for parNonEmptyTraverse, but it's okay either way

File: modules/core/src/main/scala/sup/data/HealthReporter.scala (hunk 1 above)
Comment (ragb) [reply modules/core/src/main/scala/sup/data/HealthReporter.scala]: using nonEmptyPar now.

File: modules/core/src/main/scala/sup/data/HealthReporter.scala (hunk 2 above)
Comment (ragb) [reply modules/core/src/main/scala/sup/data/HealthReporter.scala]: couldn't make it compile... or it doesn't exist or compiler is not getting some implicit.

Comment (kubukoz) [reply modules/core/src/main/scala/sup/data/HealthReporter.scala]: I'll handle it :)

**Comment by ragb** [comment]
Thanks. Learned stuff from the constraint changes you did, very helpful.

**Comment by kubukoz** [comment]
You're welcome and thanks for the contribution! Feel free to suggest any more changes/issues if you feel something is missing .

**Pull request merged**